src/csm_testing/lib/iscsi_common.py

```diff
@@ -0,0 +1,289 @@
+    Returns:
+        Worker Nodes List (list)
+    """
+    #Get List of configured worker nodes
+    sat_status_output, returncode = run_command(
+        command="""sat status --filter 'SubRole=Worker and \
+               State=Ready and \"Configuration Status\"=configured ' \
+               --fields xname  --no-borders --no-headings"""
+    )
+    if returncode != 0:
+        print(f"ERROR: Command sat status returned error. Return code: {returncode}")
+        iscsi_prechecks_failed()
+    elif sat_status_output == "":
+        print("ERROR: No Worker Node Found")
+        iscsi_prechecks_failed()
+    else:
+        #check if atleast 2 worker nodes are present
+        worker_node_list = [xname.strip() for xname in sat_status_output.split("\n")]
+        worker_node_list = [
+            xname for xname in worker_node_list if check_ssh_xname(xname)
+        ]
+        if len(worker_node_list) < 2:
+            print(
+                "ERROR: Required atleast 2 iSCSI target/server (worker node) nodes"
+            )
+            iscsi_prechecks_failed()
+        print("INFO: SSH to all worker nodes is successful.")
+    return worker_node_list
+
+def check_cli_minimum_version_xname(cli_name, minimum_version, xname):
+    """
+    Function to check cli version and print the info.
+    Args:
+        cli_name(str),minimum_version(int/float)
+    Returns:
+        None
+    """
+    cli_version_command = f"rpm -qa | grep {cli_name} | sed 's/^[a-zA-Z.-]*-\\([0-9][^ ]*\\)-.*$/\\1/'"  # pylint: disable=line-too-long
+    ssh_command = f'ssh -o BatchMode=yes -q  root@{xname} "{cli_version_command}" '
+    current_cli_version, returncode = run_command(ssh_command)
+    if returncode != 0:
+        print(f"ERROR: Failed to get {cli_name} version. Return code: {returncode}")
+        iscsi_sbps_failed(xname)
+    else:
+        # Now clean up the version by removing the package name prefix
+        # Regex to extract the version part from the package name
+        matched_cli_version = re.search(r"(\d+\.\d+\.\d+)", current_cli_version)
+
+        if matched_cli_version:
+            # Now change version to required format version in compare_versions
+            cleaned_cli_version = matched_cli_version.group(1)
+
+            if not cleaned_cli_version:
+                print(f"ERROR: {cli_name} rpm is not installed")
+                iscsi_sbps_failed(xname)
+            else:
+                if compare_versions(cleaned_cli_version, minimum_version):
+                    print(
+                        f"INFO: {cli_name} rpm version {cleaned_cli_version} >= {minimum_version}"
+                    )
+                else:
+                    print(
+                        f"ERROR: {cli_name} rpm version {cleaned_cli_version} < {minimum_version}"
+                    )
+                    iscsi_sbps_failed(xname)
+        else:
+            print("ERROR: Could not extract version from {cli_name} rpm")
+            iscsi_sbps_failed(xname)
```

Comment:
Like the previous function, this one has a lot of unnecessary indentation that makes it harder to read. It can be simplified with some very minor refactoring (no logic changes):
```suggestion
    # Now clean up the version by removing the package name prefix
    # Regex to extract the version part from the package name
    matched_cli_version = re.search(r"(\d+\.\d+\.\d+)", current_cli_version)

    if not matched_cli_version:
        print("ERROR: Could not extract version from {cli_name} rpm")
        iscsi_sbps_failed(xname)

    # Now change version to required format version in compare_versions
    cleaned_cli_version = matched_cli_version.group(1)

    if not cleaned_cli_version:
        print(f"ERROR: {cli_name} rpm is not installed")
        iscsi_sbps_failed(xname)

    if not compare_versions(cleaned_cli_version, minimum_version):
        print(f"ERROR: {cli_name} rpm version {cleaned_cli_version} < {minimum_version}")
        iscsi_sbps_failed(xname)
        
    print(f"INFO: {cli_name} rpm version {cleaned_cli_version} >= {minimum_version}")                
```